### DISCUSSION TITLE
Update without starting Comfyui.

### DISCUSSION THREAD

**Discussion by razvanab** [Comfy-Org/ComfyUI-Manager#710] [Ideas]
Can you update Comfyui and the nodes without starting Comfyui, update them, and then restart Comfyui again?

I used to do this with a PS script to start git sync on everything, but that doesn't start the installation of the requirements.txt, if there were any.

**Comment by ltdrdata** [comment]
There is `cm-cli.py`.
https://github.com/ltdrdata/ComfyUI-Manager/blob/main/docs/en/cm-cli.md

And I am working on `comfy-cli` project.
https://github.com/yoland68/comfy-cli

If you are using an environment like venv, I highly recommend `comfy-cli`.
I also always use the `comfy update all` command for update routines lately.

**Reply by razvanab** [reply]
Nice I didn't know about this.
Thank you.

**Comment by razvanab** [comment]
Oh, one more thing. 
I think I read that right. Can you export as a config file the current list of installed nodes? 
So next, you won't need to reinstall them again by selecting them from the UI list?

**Reply by ltdrdata** [reply]
There is several ways. 
1) You can `save` & `restore` on snapshot menu.
2) And another one is `cm-cli.py`. The command line interface providing such action.
3) `comfy-cli` supports snapshot feature as well.
`comfy node save-snapshot --output=<file>`
`comfy node restore-snapshot <file>`

**Reply by razvanab** [reply]
It is good to know.
Thank you
